Gordon360/Services/ActivityService.cs

```diff
@@ -308,6 +312,7 @@ public async Task<ACT_INFO> UpdateActivityImageAsync(ACT_INFO involvement, IForm
             // Put current DateTime in filename so the browser knows it's a new file and refreshes cache
             var filename = $"canvasImage_{DateTime.Now:yyyy-MM-dd_HH-mm-ss-fff}.png";
             var imagePath = Path.Combine(_webHostEnvironment.ContentRootPath, "browseable", "uploads", involvement.ACT_CDE.Trim(), filename);
+            var url = $"{ _server.Features.Get<IServerAddressesFeature>()?.Addresses.FirstOrDefault(a => a.Contains("https"))}/browseable/uploads/{involvement.ACT_CDE.Trim()}/{filename}";
```

Comment:
I think that we could make this code a fair bit cleaner. Instead of using the raw `IServer` object, we might want to create a `ServerUtils` class that exists as a Scoped dependency and wraps `IServer`.

We could then call `ServerUtils.GetAddress`, which would return the best current server base URL. That method could also include some logic to return the `https` address if it exists, or else the first address.

Then, this line would simplify to:
```suggestion
            var imageUrl = $"{ _serverUtils.GetAddress()}/browseable/uploads/{involvement.ACT_CDE.Trim()}/{filename}";
```

(Also, you could extract `involvement.ACT_CDE.Trim()` to a local variable so that we trim it only once, rather than both times it's referenced.)

Reply:
@sharonpatta did you consider this one? Let me know if you want to set up a time soon to work on this together. It would be good to familiarize yourself with the idea of dependency injection.